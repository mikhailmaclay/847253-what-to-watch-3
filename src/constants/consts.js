export const Config = {
  MOVIE_CARDS_COUNT_AT_START: 8,
  MOVIE_CARDS_COUNT_BY_LOAD: 8,
  MOVIE_SCORE_PRECISION: 3,
  MOVIE_LEVEL_MAP: {
    10: `Awesome`,
    7: `Very good`,
    5: `Good`,
    3: `Normal`,
    0: `Bad`
  },
  MOVIE_PROMO_RELEASE_DATE_FORMAT: `MMMM DD, YYYY`,
  MOVIE_DETAILS_RELEASE_DATE_FORMAT: `YYYY`,
  MOVIE_DETAILS_RUN_TIME_FORMAT: `H[h] m[m]`,
  REVIEW_DATE_FORMAT: `MMMM DD, YYYY`,
  MOVIE_DEFAULT_SCORE: 0,
  MOVIE_OVERVIEW_ACTORS_COUNT: 4,
  SIMILAR_MOVIES_COUNT: 4,
  VIDEO_PLAYER_PLAY_TIME_FORMAT: `HH:mm:ss`,
  VIDEO_PLAYER_DEFAULT_VOLUME: 1,
  VIDEO_PLAYER_IS_MUTED_ON_START: false,
  VIDEO_PLAYER_IS_PLAYING_ON_START: true,
  MOVIE_PREVIEW_DELAY_MS: 1000,
  MOVIE_PREVIEW_DEFAULT_VOLUME: 1,
  MOVIE_PREVIEW_IS_LOOPED: true,
  MOVIE_PREVIEW_IS_MUTED_ON_START: true,
  MOVIE_PREVIEW_IS_PLAYING_ON_START: true,
  GENRE_LIST_MAX_MOVIE_COUNT: 9,
  REQUEST_TIMEOUT: 5000,
  NOTIFICATION_TIMEOUT: 3000,
  MIN_REVIEW_LENGTH: 50,
  MAX_REVIEW_LENGTH: 400
};

export const NO_SPECIAL_MOVIE = -1;

export const PathName = {
  ROOT: `/`,
  MOVIE_FILTER: `/films`,
  MOVIE_PAGE: `/films/`,
  WATCH: `/watch/`,
  SIGN_IN: `/sign-in`,
  MY_LIST: `/my-list`,
  SERVER: `https://htmlacademy-react-3.appspot.com`
};

export const FilterName = {
  MOVIES: {
    ID: `FILTER_MOVIE_ID`,
    EXCEPT_ID: `FILTER_MOVIE_EXCEPT_ID`,
    GENRE: `FILTER_MOVIE_GENRE`
  },
  TEAM: {
    ROLE: `FILTER_TEAM_ROLE`
  },
  REVIEWS: {
    ID: `FILTER_REVIEW_ID`
  },
  USERS: {
    ID: `FILTER_USER_ID`
  }
};

export const SelectorName = {
  MOVIES: {
    GENRES: `SELECTOR_MOVIES_GENRES`
  },
  MOVIE: {
    TEAM: `SELECTOR_MOVIE_TEAM`,
    RATINGS: `SELECTOR_MOVIE_RATINGS`
  }
};

export const ConverterName = {
  RATING: {
    LEVEL: `CONVERTER_RATING_LEVEL`
  }
};

export const AdapterName = {
  MOVIE: `ADAPTER_MOVIE`
};

export const KeyCode = {
  ESC: `Escape`
};

export const AuthorizationStatus = {
  AUTHORIZED: `AUTHORIZED`,
  UNAUTHORIZED: `UNAUTHORIZED`
};
