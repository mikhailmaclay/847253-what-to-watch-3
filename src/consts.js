export const Config = {
  SPECIAL_MOVIE_ID: 1,
  MOVIE_CARDS_COUNT_BY_LOAD: 8,
  MOVIE_SCORE_PRECISION: 3,
  MOVIE_LEVEL_MAP: {
    10: `Awesome`,
    7: `Very good`,
    5: `Good`,
    3: `Normal`,
    0: `Bad`
  },
  MOVIE_DEFAULT_SCORE: 0,
  MOVIE_OVERVIEW_ACTORS_COUNT: 4,
  SIMILAR_MOVIES_COUNT: 4
};

export const PathName = {
  ROOT: `/`,
  MOVIE_FILTER: `/films`,
  MOVIE_PAGE: `/films/`,
  ADD_REVIEW: `/add-review`
};

export const FilterName = {
  Movie: {
    EXCEPT_ID: `ID`,
    GENRE: `GENRE`
  },
  Team: {
    ROLE: `ROLE`
  }
};

