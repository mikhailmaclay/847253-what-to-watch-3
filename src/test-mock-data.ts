import {User, Movie, Review, TeamMember, Notification} from './types';

export const mockUser: User = {
  id: 1,
  email: `JohnnyMnemonyc@mail.ru`,
  name: `Johnny`,
  avatar: `https://htmlacademy-react-3.appspot.com/wtw/static/avatar/4.jpg`,
  myList: null
};

export const mockReview: Review = {
  id: 1,
  user: {id: mockUser.id, fullName: mockUser.name},
  date: `2020-03-24T08:26:50.139Z`,
  rating: 10,
  text: `It was really awesome!`
};

export const mockDirector: TeamMember = {
  fullName: `Wes Anderson`,
  role: `Director`
};

export const mockActor: TeamMember = {
  fullName: `Adrien Brody`,
  role: `Actor`
};

export const mockMovie: Movie = {
  id: 1,
  name: `Dardjeeling Limited`,
  genre: `Adventure`,
  releaseDate: `2007`,
  description: [
    `A year after their father\`s funeral, three brothers travel across India by train in an attempt to bond with each other.`
  ],
  runTime: 91,
  team: [
    mockDirector,
    mockActor
  ],
  poster: `https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Dardjeeling_Limited.jpg`,
  images: [
    `https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/dardjeeling_limited.jpg`
  ],
  backgrounds: [
    `#AD9F8B`,
    `https://htmlacademy-react-3.appspot.com/wtw/static/film/background/Dardjeeling_Limited.jpg`
  ],
  rating: 3.6,
  scoresCount: 165106,
  preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  video: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
  isInMyList: false,
  reviews: null
};

export const mockNotification: Notification = {
  title: `Test`,
  content: `Testing message.`
};
