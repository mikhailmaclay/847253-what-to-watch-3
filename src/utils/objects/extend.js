function extend(...objects) {
  const temp = {};

  objects.forEach((object) => Object.assign(temp, object));

  return temp;
}

export default extend;
