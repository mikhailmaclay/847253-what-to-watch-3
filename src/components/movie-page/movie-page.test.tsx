// Libraries
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import {Router} from 'react-router-dom';
// Components
import MoviePage from './movie-page';
//
import history from '../../history';
import {PathName} from '../../constants/consts';
import emptyArrowFunction from '../../utils/components/empty-arrow-function.js';
import extend from '../../utils/objects/extend';
import {mockMovie} from '../../test-mock-data';
import {mockActor, mockDirector} from '../../test-mock-data';

const FIRST_MOVIE_INDEX = 0;

const mockStore = configureStore([]);

describe(`<MoviePage/>`, () => {
  const store = mockStore({
    movies: [extend(mockMovie, {
      level: `Awesome`,
      directors: [mockDirector.fullName],
      actors: [mockActor.fullName],
      similarMovies: []
    })]
  });

  const props = {
    movie: store.getState().movies[FIRST_MOVIE_INDEX],
    baseURL: PathName.MOVIE_PAGE + store.getState().movies[FIRST_MOVIE_INDEX].id,
    isAuthorized: false,
    onChangeMovieStatus: emptyArrowFunction
  };

  it(`Should render correctly`, () => {
    const result = renderer.create(
        <Provider store={store}>
          <Router history={history}>
            <MoviePage {...props}/>
          </Router>
        </Provider>
    ).toJSON();

    expect(result).toMatchSnapshot();
  });
});
