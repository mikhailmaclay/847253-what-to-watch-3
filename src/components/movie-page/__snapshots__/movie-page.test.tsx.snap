// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MoviePage/> Should render correctly 1`] = `
Array [
  <section
    className="movie-card movie-card--full"
  >
    <div
      className="movie-card__hero"
    >
      <div
        className="movie-card__bg"
        style={
          Object {
            "backgroundColor": "#AD9F8B",
          }
        }
      >
        <img
          alt="Dardjeeling Limited"
          src="https://htmlacademy-react-3.appspot.com/wtw/static/film/background/Dardjeeling_Limited.jpg"
        />
      </div>
      <h1
        className="visually-hidden"
      >
        WTW
      </h1>
      <header
        className="page-header movie-card__head"
      >
        <div
          className="logo"
        >
          <a
            className="logo__link "
            href={null}
            onClick={null}
          >
            <span
              className="logo__letter logo__letter--1"
            >
              W
            </span>
            <span
              className="logo__letter logo__letter--2"
            >
              T
            </span>
            <span
              className="logo__letter logo__letter--3"
            >
              W
            </span>
          </a>
        </div>
        <div
          className="user-block"
        >
          <a
            className="user-block__link"
            href="/login"
            onClick={[Function]}
          >
            Sign in
          </a>
        </div>
      </header>
      <div
        className="movie-card__wrap"
      >
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Dardjeeling Limited
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Adventure
            </span>
            <span
              className="movie-card__year"
            >
              January 01, 2007
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <a
              className="btn btn--play movie-card__button"
              href="/films/1/player"
              onClick={[Function]}
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </a>
            <button
              className="btn btn--list movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="movie-card__wrap movie-card__translate-top"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster movie-card__poster--big"
        >
          <img
            alt="Dardjeeling Limited"
            height="327"
            src="https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Dardjeeling_Limited.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <nav
            className="movie-nav movie-card__nav"
          >
            <ul
              className="movie-nav__list"
            >
              <li
                className="movie-nav__item "
              >
                <a
                  className="movie-nav__link"
                  href="/films/1"
                  onClick={[Function]}
                  style={
                    Object {
                      "pointerEvents": "",
                      "userSelect": "none",
                    }
                  }
                >
                  Overview
                </a>
              </li>
              <li
                className="movie-nav__item "
              >
                <a
                  className="movie-nav__link"
                  href="/films/1/details"
                  onClick={[Function]}
                  style={
                    Object {
                      "pointerEvents": "",
                      "userSelect": "none",
                    }
                  }
                >
                  Details
                </a>
              </li>
              <li
                className="movie-nav__item "
              >
                <a
                  className="movie-nav__link"
                  href="/films/1/reviews"
                  onClick={[Function]}
                  style={
                    Object {
                      "pointerEvents": "",
                      "userSelect": "none",
                    }
                  }
                >
                  Reviews
                </a>
              </li>
            </ul>
          </nav>
          <div
            className="movie-rating"
          >
            <div
              className="movie-rating__score"
            >
              3.6
            </div>
            <p
              className="movie-rating__meta"
            >
              <span
                className="movie-rating__level"
              >
                Awesome
              </span>
              <span
                className="movie-rating__count"
              >
                165106 ratings
              </span>
            </p>
          </div>
          <div
            className="movie-card__text"
          >
            <p>
              A year after their father\`s funeral, three brothers travel across India by train in an attempt to bond with each other.
            </p>
            <p
              className="movie-card__director"
            >
              <b>
                Director
                : 
                Wes Anderson
              </b>
            </p>
            <p
              className="movie-card__starring"
            >
              <b>
                Starring: 
                Adrien Brody
              </b>
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href={null}
          onClick={null}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
